instrumentation/jedis-5.0.0/src/main/java/redis/clients/jedis/Connection_Instrumentation.java

```diff
@@ -0,0 +1,90 @@
+/*
+ *
+ *  * Copyright 2020 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package redis.clients.jedis;
+
+import com.newrelic.agent.bridge.AgentBridge;
+import com.newrelic.agent.bridge.datastore.DatastoreVendor;
+import com.newrelic.api.agent.DatastoreParameters;
+import com.newrelic.api.agent.NewRelic;
+import com.newrelic.api.agent.Trace;
+import com.newrelic.api.agent.weaver.MatchType;
+import com.newrelic.api.agent.weaver.NewField;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+import redis.clients.jedis.commands.ProtocolCommand;
+
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Level;
+
+@SuppressWarnings({ "ResultOfMethodCallIgnored", "WeakerAccess", "unused" }) // Weaver.callOriginal(), matching signatures
+@Weave(type = MatchType.BaseClass, originalName = "redis.clients.jedis.Connection")
+public abstract class Connection_Instrumentation {
+
+    @NewField
+    private long db = 0;
+
+    abstract HostAndPort getHostAndPort();
+
+    public void disconnect() {
+        db = 0;
+        Weaver.callOriginal();
+    }
+
+    @Trace
+    public void sendCommand(final ProtocolCommand cmd, final byte[]... args) {
+        Weaver.callOriginal();
+        if (args != null && args.length > 0) {
+            updateDbIndex(cmd, new String(args[0], StandardCharsets.UTF_8));
+        }
+    }
+
+    @Trace
+    public void sendCommand(final ProtocolCommand cmd, final String... args) {
+        Weaver.callOriginal();
+        if (args != null && args.length > 0) {
+            updateDbIndex(cmd, args[0]);
+        }
+    }
+
+    @Trace(leaf = true)
```

Comment:
What is the reasoning behind this method being a leaf and the others not. It would be good to document so later we are not guessing.